k8s/controllers/workflow/activation_controller.go

```diff
@@ -84,6 +85,8 @@ func (r *ActivationReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 				return ctrl.Result{}, err
 			}
 		}
+	} else {
+		observ_utils.EmitUserAuditsLogs(ctx, "Activation %s is being deleted on namespace %s", activation.ObjectMeta.Name, activation.ObjectMeta.Namespace)
```

Comment:
You need below below lines

```go
// construct a context with required properties for user logs
ctx := configutils.PopulateActivityAndDiagnosticsContextFromAnnotations(r.GetNamespace(), resourceK8SId, r.Annotations, operationName, myActivationClient, context.TODO(), activationlog)
// pod logs
diagnostic.InfoWithCtx(activationlog, ctx, "Deleting activation", "name", r.Name, "namespace", r.Namespace)
observ_utils.EmitUserAuditsLogs(ctx, "Activation %s is being created on namespace %s", r.Name, r.Namespace)
```

Reply:
Thank you for providing those example code

Reply:
I took a look at the controller logs, better to iterate all controllers and add context construction before calling `diagnostic.InfoWithCtx`. Previously, we only do them on solution deployment (in k8s/reconcilers/deployment.go). Given we have more and more controller logs, let's do them at controller side.